usaspending_api/search/v2/views/new_awards_over_time.py

```diff
@@ -28,93 +29,104 @@ class NewAwardsOverTimeVisualizationViewSet(APIView):
+                recipient_hash=recipient_hash, recipient_level='P'
+            ).values('recipient_unique_id')
             if len(parent_duns_rows) != 1:
-                raise InvalidParameterException("Provided recipient_id has no parent records")
-            parent_duns = parent_duns_rows[0]["recipient_unique_id"]
+                raise InvalidParameterException('Provided recipient_id has no parent records')
+            parent_duns = parent_duns_rows[0]['recipient_unique_id']
             queryset = queryset.filter(parent_recipient_unique_id=parent_duns)
-        elif self.filters["recipient_id"][-1] == "R":
+        elif self.filters['recipient_id'][-1] == 'R':
             queryset = queryset.filter(recipient_hash=recipient_hash, parent_recipient_unique_id__isnull=True)
         else:
             queryset = queryset.filter(recipient_hash=recipient_hash, parent_recipient_unique_id__isnull=False)
 
-        values = ["fy"]
-        if self.groupings[self.json_request["group"]] == "month":
-            queryset = queryset.annotate(month=FiscalMonth("date_signed"), fy=FiscalYear("date_signed"))
-            values.append("month")
+        values = ['fy']
+        if self.groupings[self.json_request['group']] == 'month':
+            queryset = queryset.annotate(month=FiscalMonth('action_date'), fy=FiscalYear('action_date'))
+            values.append('month')
 
-        elif self.groupings[self.json_request["group"]] == "quarter":
-            queryset = queryset.annotate(quarter=FiscalQuarter("date_signed"), fy=FiscalYear("date_signed"))
-            values.append("quarter")
+        elif self.groupings[self.json_request['group']] == 'quarter':
+            queryset = queryset.annotate(quarter=FiscalQuarter('action_date'), fy=FiscalYear('action_date'))
+            values.append('quarter')
 
-        elif self.groupings[self.json_request["group"]] == "fiscal_year":
-            queryset = queryset.annotate(fy=FiscalYear("date_signed"))
+        elif self.groupings[self.json_request['group']] == 'fiscal_year':
+            queryset = queryset.annotate(fy=FiscalYear('action_date'))
 
-        queryset = (
-            queryset.values(*values).annotate(count=Sum("counts")).order_by(*["-{}".format(value) for value in values])
-        )
+        queryset = queryset.values(*values).annotate(count=Count('award_id'))
         return queryset, values
 
     @cache_response()
     def post(self, request):
         self.json_request = self.validate_api_request(request.data)
-        self.filters = self.json_request.get("filters", None)
+        self.filters = self.json_request.get('filters', None)
 
         if self.filters is None:
-            raise InvalidParameterException("Missing request parameters: filters")
+            raise InvalidParameterException('Missing request parameters: filters')
 
         queryset, values = self.database_data_layer()
 
-        results = []
+        # Populate all possible periods with no new awards for now
```

Comment:
1. It might be a good idea to use a new API parameter to keep the response as-is or to populate all of the missing periods.

2. Can this code and the code in spending_over_time be moved to a shared helper file and imported instead of duplicated?

Reply:
Point 1 boils down to _adding new functionality_ rather than changing the behavior/response of an existing API endpoint.

Reply:
Point 2 has been addressed. Point 1 is setup though I would want to discuss with the frontend team as it could affect the contract for this endpoint.